compiler/include/baseAST.h

```diff
@@ -392,6 +398,25 @@ static inline const LcnSymbol* toConstLcnSymbol(const BaseAST* a)
   return isLcnSymbol(a) ? (const LcnSymbol*) a : NULL;
 }
 
+CallExpr* getDesignatedCall(ContextCallExpr* a);
+
+static inline CallExpr* toCallExpr(BaseAST* a)
+{
+  if (!a) return NULL;
+  if (a->astTag == E_CallExpr) return (CallExpr*) a;
+  if (a->astTag == E_ContextCallExpr) return getDesignatedCall((ContextCallExpr*)a);
+  return NULL;
+}
+
+static inline const CallExpr* toConstCallExpr(const BaseAST* a)
+{
+  if (!a) return NULL;
+  if (a->astTag == E_CallExpr) return (const CallExpr*) a;
+  if (a->astTag == E_ContextCallExpr) return getDesignatedCall((ContextCallExpr*)a);
+  return NULL;
+}
```

Comment:
Well, two things. First, there isn't necessarily a const ref call since it could return by value. Second, I was just providing the same functions that def_to_ast does. That macro provides to##Type and toConst##Type. The const version is just referring to C++ const and has nothing to do with Chapel semantics.